os/arch/arm/src/imxrt/imxrt_ehci.c

```diff
@@ -0,0 +1,5140 @@
+#define CONFIG_IMXRT_EHCI_BUFSIZE 128
+#endif
+
+#define CONFIG_IMXRT_EHCI_PREALLOCATE
+
+#define IMXRT_EHCI_BUFSIZE \
+	((CONFIG_IMXRT_EHCI_BUFSIZE + DCACHE_LINEMASK) & ~DCACHE_LINEMASK)
+
+/* Debug options */
+
+#ifndef CONFIG_DEBUG_USB_INFO
+#undef CONFIG_IMXRT_EHCI_REGDEBUG
+#endif
+
+/* Isochronous transfers are not currently supported */
+
+#undef CONFIG_USBHOST_ISOC_DISABLE
+#define CONFIG_USBHOST_ISOC_DISABLE 1
+
+/* Registers *******************************************************************/
+/* Traditionally, NuttX specifies register locations using individual
+ * register offsets from a base address.  That tradition is broken here and,
+ * instead, register blocks are represented as structures.  This is done here
+ * because, in principle, EHCI operational register address may not be known
+ * at compile time; the operational registers lie at an offset specified in
+ * the 'caplength' byte of the Host Controller Capability Registers.
+ *
+ * However, for the case of the LPC31 EHCI, we know apriori that locations
+ * of these register blocks.
+ */
+
+/* Host Controller Capability Registers */
+
+#define HCCR ((struct ehci_hccr_s *)IMXRT_USBOTG_HCCR_BASE)
+
+/* Host Controller Operational Registers */
+
+#define HCOR ((volatile struct ehci_hcor_s *)IMXRT_USBOTG_HCOR_BASE)
+
+/* Interrupts ******************************************************************/
+/* This is the set of interrupts handled by this driver */
+
+#define EHCI_HANDLED_INTS (EHCI_INT_USBINT | EHCI_INT_USBERRINT | \
+							EHCI_INT_PORTSC |  EHCI_INT_SYSERROR | \
+							EHCI_INT_AAINT)
+
+/* The periodic frame list is a 4K-page aligned array of Frame List Link
+ * pointers. The length of the frame list may be programmable. The programmability
+ * of the periodic frame list is exported to system software via the HCCPARAMS
+ * register. If non-programmable, the length is 1024 elements. If programmable,
+ * the length can be selected by system software as one of 256, 512, or 1024
+ * elements.
+ */
+
+#define FRAME_LIST_SIZE 1024
+
+/* DMA *************************************************************************/
+/* For now, we are assuming an identity mapping between physical and virtual
+ * address spaces.
+ */
+
+#define imxrt_physramaddr(a) (a)
+#define imxrt_virtramaddr(a) (a)
+
+/* USB trace *******************************************************************/
+
+#ifdef HAVE_USBHOST_TRACE
+#define TR_FMT1 false
+#define TR_FMT2 true
+
+#define TRENTRY(id, fmt1, string) {string}
+
+#define TRACE1_FIRST     ((int)__TRACE1_BASEVALUE + 1)
+#define TRACE1_INDEX(id) ((int)(id) - TRACE1_FIRST)
+#define TRACE1_NSTRINGS  TRACE1_INDEX(__TRACE1_NSTRINGS)
+
+#define TRACE2_FIRST     ((int)__TRACE1_NSTRINGS + 1)
+#define TRACE2_INDEX(id) ((int)(id) - TRACE2_FIRST)
+#define TRACE2_NSTRINGS  TRACE2_INDEX(__TRACE2_NSTRINGS)
+#endif
+
+/* Port numbers */
+
+#define RHPNDX(rh)            ((rh)->hport.hport.port)
+#define RHPORT(rh)            (RHPNDX(rh)+1)
```

Comment:
please provide space before and after '+'

Reply:
I fixed it